### ISSUE TITLE
 Unexpected token error in react-native-reanimated during iOS bundle building

### ISSUE THREAD

**Issue by alexander0205** [software-mansion/react-native-reanimated#4835]
### Description

While attempting to build the bundle for iOS platform using npx react-native bundle, an unexpected token error occurred in the react-native-reanimated module. The error is related to an unexpected operator (=) token in node_modules/react-native-reanimated/src/reanimated2/animation/util.ts at line 153, position 21.

`npx react-native bundle --entry-file index.js --platform ios --dev false --bundle-output ios/main.jsbundle --assets-dest ios`

```
error node_modules/react-native-reanimated/src/reanimated2/animation/util.ts: Unexpected token: operator (=) in file node_modules/react-native-reanimated/src/reanimated2/animation/util.ts at 153:21.
Error: Unexpected token: operator (=) in file node_modules/react-native-reanimated/src/reanimated2/animation/util.ts at 153:21
    at minifyCode (/Volumes/ExternalHD/Nippy/nippy-mobile2/node_modules/metro-transform-worker/src/index.js:99:13)
    at transformJS (/Volumes/ExternalHD/Nippy/nippy-mobile2/node_modules/metro-transform-worker/src/index.js:320:28)
    at transformJSWithBabel (/Volumes/ExternalHD/Nippy/nippy-mobile2/node_modules/metro-transform-worker/src/index.js:411:16)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async Object.transform (/Volumes/ExternalHD/Nippy/nippy-mobile2/node_modules/metro-transform-worker/src/index.js:572:12)
info Run CLI with --verbose flag for more details.
```

### Steps to reproduce

1. Run this code in the project 
2. `npx react-native bundle --entry-file index.js --platform ios --dev false --bundle-output ios/main.jsbundle --assets-dest ios`

### Snack or a link to a repository

https://snack.expo.dev/

### Reanimated version

3.0.1

### React Native version

0.70.6

### Platforms

iOS

### JavaScript runtime

None

### Workflow

None

### Architecture

None

### Build type

None

### Device

None

### Acknowledgements

Yes

**Comment by alexander0205** [comment]
npx react-native bundle --entry-file index.js --platform ios --dev false --bundle-output ios/main.jsbundle --assets-dest ios

> Hey! 👋
> 
> The issue doesn't seem to contain a [minimal reproduction](https://stackoverflow.com/help/minimal-reproducible-example).
> 
> Could you provide a [snack](https://snack.expo.dev/) or a link to a GitHub repository under your username that reproduces the problem?

**Comment by tjzel** [comment]
Hi @alexander0205, what version of `node` do you use? Also, when you look into [npm published code](https://www.npmjs.com/package/react-native-reanimated/v/3.0.1?activeTab=code) of the file that is troublesome here you can notice that there is no `=` on the offending line. This might be some parsing error and line numbers got shifted somehow but could you double check if you you are using Reanimated 3.0.1 (e.g. by looking into `node_modules/react-native-reanimated/package.json`)?

![image](https://github.com/software-mansion/react-native-reanimated/assets/40713406/ebeba77a-9b4f-47eb-8fd0-6648d1e95773)

**Comment by alexander0205** [comment]
> Hi @alexander0205, what version of `node` do you use?
> 
> Also, when you look into [npm published code](https://www.npmjs.com/package/react-native-reanimated/v/3.0.1?activeTab=code) of the file that is troublesome here you can notice that there is no `=` on the offending line. This might be some parsing error and line numbers got shifted somehow but could you double check if you you are using Reanimated 3.0.1 (e.g. by looking into `node_modules/react-native-reanimated/package.json`)?
> 
> ![image](https://user-images.githubusercontent.com/40713406/256584308-ebeba77a-9b4f-47eb-8fd0-6648d1e95773.png)

@tjzel I try with node 14,16,18

**Comment by alexander0205** [comment]
The error was resolved by using version 3.3.0 of react-native-reanimated. This issue only occurred with versions 3.4.0 and 3.4.1.

**Comment by tjzel** [comment]
Therefore it's duplicate of #4825
